rtpengine: rtpengine_allow_op return node logic

descartin opened the pull request:
When using rtpengine_allow_op param, return node for ongoing calls only if rtpengine is manually disabled.
 
When a rtpengine instance is disabled, rtpengine_allow_op will allow the indialog requests to be processed through that rtpengine. 
But it's not desirable to return the node when the rtpengine has unexpectedly crashed or has any undesired issues, because the rtp commands attempted cause congestion problem in the kamailio.
So with this patch we only return the node if the rtpengine has been manually disabled and it's still supposed to be working.


<!-- Kamailio Pull Request Template -->

<!--
IMPORTANT:
  - for detailed contributing guidelines, read:
    https://github.com/kamailio/kamailio/blob/master/.github/CONTRIBUTING.md
  - pull requests must be done to master branch, unless they are backports
    of fixes from master branch to a stable branch
  - backports to stable branches must be done with 'git cherry-pick -x ...'
  - code is contributed under BSD for core and main components (tm, sl, auth, tls)
  - code is contributed GPLv2 or a compatible license for the other components
  - GPL code is contributed with OpenSSL licensing exception
-->

#### Pre-Submission Checklist
<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask on sr-dev mailing list -->
- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [ ] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ x] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:
<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply -->
- [x ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description
<!-- Describe your changes in detail -->

henningw commented:
Looks good to me. From the documentation this feature should work like this in the first place.

But I did a review of the code, and it seems this feature was modified to actually work in the (in your opinion not desired way), but probably the docs not adapted. See below.

So probably a better idea is to introduce another mode "2" for the rtpengine_allow_op parameter which then trigger the desired behaviour.

> commit 5a537506141027ca3d3ef87f49913ab628c30690
> Author: Stefan Mititelu <stefan.mititelu@1and1.ro>
> Date:   Mon Dec 7 13:30:02 2015 +0200
> 
>     rtpengine: Allow op for all deactivated machines
> 
>     If allow_op modparam enabled, send commands to the disabled machines for the
>     existing call. So far this was done only for manually deactivated machines.
>     This is useful because there might be cases of proxy timeout, cases when you
>     may want to still allow the operations for the existing calls.
>

descartin commented:
Hello Henning

thanks for the feedback, we will study the case to create a new patch
including the different modes

best regards
david

El lun, 12 jul 2021 a las 15:08, 'Henning Westerholt' via descartin (<
***@***.***>) escribió:

> Looks good to me. From the documentation this feature should work like
> this in the first place.
>
> But I did a review of the code, and it seems this feature was modified to
> actually work in the (in your opinion not desired way), but probably the
> docs not adapted. See below.
>
> So probably a better idea is to introduce another mode "2" for the
> rtpengine_allow_op parameter which then trigger the desired behaviour.
>
> commit 5a53750
> <https://github.com/kamailio/kamailio/commit/5a537506141027ca3d3ef87f49913ab628c30690>
> Author: Stefan Mititelu ***@***.***
> Date: Mon Dec 7 13:30:02 2015 +0200
>
> rtpengine: Allow op for all deactivated machines
>
> If allow_op modparam enabled, send commands to the disabled machines for the
> existing call. So far this was done only for manually deactivated machines.
> This is useful because there might be cases of proxy timeout, cases when you
> may want to still allow the operations for the existing calls.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kamailio/kamailio/pull/2800#issuecomment-878264302>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB3IPOX3K6A2TVA7VWBKI3TTXLSLTANCNFSM5AGONYGQ>
> .
>


-- 
[image: Logo]

David Escartín Almudévar
VoIP/Switch Engineer
***@***.***

*SONOC*
C/ Josefa Amar y Borbón, 10, 4ª · 50001 Zaragoza, España
Tlf: +34 917019888 · www.sonoc.io

David Escartín
NOC engineer
www.sonoc.io
This e-mail is for the exclusive use of its recipients and may contain business secrets or other confidential or privileged information. Any unauthorised use, copying, publication or distribution of this e-mail is strictly prohibited. If you are not the intended recipient, please inform us immediately by replying to this e-mail and delete it, including any attachments or copies on your system.
​
In accordance with the GDPR (EU) 2016/679 and the LOPDGDD 3/2018, we inform you that this e-mail address and/or any other personal data you have provided us with will be treated by SONOC with absolute confidentiality and with the only purpose of providing you with the requested services, due to your condition as a client, supplier or because you have requested information from us at any time. These data will only be kept for as long as required to comply with legal obligations. You can exercise your rights at any time by sending an e-mail to: ***@***.***

henningw commented:
Thanks for the update. @rfuchs any comments on this PR? Otherwise I think it could be merged.

descartin commented:
hello all

we have added a new commit to add mode 2 for rtpengine_allow_op param to return only manually disabled nodes
We updated the documentation too.

regards

rfuchs commented:
LGTM overall, except that the debug log message ("return it") now doesn't necessarily reflect what it actually does :laughing:

henningw commented:
Thanks, merged. I will push another small commit to fix the commit msg for the new mode.

henningw commented:
@descartin in case of future pull request, please add your name and e-mail to your git config. Currently its setup like this:
> Author: david <you@example.com>

descartin commented:
oops, ok i didn't notice it, thank you



El mar, 13 jul 2021 a las 15:07, 'Henning Westerholt' via descartin (<
***@***.***>) escribió:

> @descartin <https://github.com/descartin> in case of future pull request,
> please add your name and e-mail to your git config. Currently its setup
> like this:
>
> Author: david ***@***.***
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kamailio/kamailio/pull/2800#issuecomment-879071440>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB3IPOXNUNKN3NHS4EV5C23TXQ27FANCNFSM5AGONYGQ>
> .
>


-- 
[image: Logo]

David Escartín Almudévar
VoIP/Switch Engineer
***@***.***

*SONOC*
C/ Josefa Amar y Borbón, 10, 4ª · 50001 Zaragoza, España
Tlf: +34 917019888 · www.sonoc.io

David Escartín
NOC engineer
www.sonoc.io
This e-mail is for the exclusive use of its recipients and may contain business secrets or other confidential or privileged information. Any unauthorised use, copying, publication or distribution of this e-mail is strictly prohibited. If you are not the intended recipient, please inform us immediately by replying to this e-mail and delete it, including any attachments or copies on your system.
​
In accordance with the GDPR (EU) 2016/679 and the LOPDGDD 3/2018, we inform you that this e-mail address and/or any other personal data you have provided us with will be treated by SONOC with absolute confidentiality and with the only purpose of providing you with the requested services, due to your condition as a client, supplier or because you have requested information from us at any time. These data will only be kept for as long as required to comply with legal obligations. You can exercise your rights at any time by sending an e-mail to: ***@***.***
